core/templates/dev/head/components/CreateActivityButtonDirective.js

```diff
@@ -40,33 +40,37 @@ oppia.directive('createActivityButton', [function() {
         };
 
         $scope.showCreationChoiceModal = function() {
-          $scope.creationInProgress = true;
-          if (urlService.getPathname() !== '/my_explorations') {
-            $window.location.replace('/my_explorations?mode=create');
-          } else {
-            $modal.open({
-              templateUrl: 'modals/createActivity',
-              backdrop: true,
-              controller: [
-                  '$scope', '$modalInstance', function($scope, $modalInstance) {
-                $scope.chooseExploration = function() {
-                  ExplorationCreationService.createNewExploration();
-                  $modalInstance.close();
-                };
+          if (!$scope.creationInProgress) {
```

Comment:
So, this works, but I do have a nit -- the way it's done here introduces quite a bit of additional nesting. Probably better would be a few lines at the top like:

```
    // Without this, the modal keeps reopening when the window is resized.
    if ($scope.creationInProgress) {
      return;
    }
```

and leave the rest of the lines as they were.